File: 1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        mod = 10**9 + 7
        str = ""

        for i in range(1, n + 1):
            b = bin(i)[2:]
            str += b

        return int(str, 2) % mod

File: 1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.cs
public class Solution {
    public int ConcatenatedBinary(int n) {
        const int MOD = 1000000007;
        long ans = 0;
        var pow = 2;

        for (int i = 1; i <= n; i++) {
            if (pow == i) {
                pow *= 2;
            }
            ans = (ans * pow + i) % MOD;
        }

        return (int)ans;
    }
}